activity_calendar/views.py

```diff
@@ -131,171 +38,292 @@ def check_join_constraints(request, parent_activity, recurrence_id):
+            'data': {
+                'sign_up': not self.activity.is_user_subscribed(self.request.user, self.recurrence_id)
+            },
+            'activity': self.activity,
+            'recurrence_id': self.recurrence_id,
+            'user': self.request.user,
+        }
+        kwargs.update(super(ActivitySimpleMomentView, self).get_form_kwargs())
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        kwargs = super(ActivitySimpleMomentView, self).get_context_data(**kwargs)
+        kwargs.update({
+            'participants': self.activity.get_subscribed_participants(recurrence_id=self.recurrence_id),
+        })
+        return kwargs
+
+
+class ActivityMomentWithSlotsView(LoginRequiredForPostMixin, FormMixin, ActivityMomentView):
+    form_class = RegisterForActivitySlotForm
+    template_name = "activity_calendar/activity_page_slots.html"
+
+    error_messages = {
+        'undefined': _("Something went wrong. Please try again."),
+        'activity-full': _("This activity is already at maximum capacity. You can not subscribe to it."),
+        'invalid': _("You can not subscribe to this activity. Reason currently undefined"),
+        'already-registered': _("You are already registered for this activity"),
+        'not-registered': _("You are not registered to this activity"),
+        'closed': _("You can not subscribe, subscriptions are currently closed"),
+        'max-slots-occupied': _("You can not subscribe to another slot. Maximum number of subscribable slots already reached"),
+        # Slot creation error messages
+        'max-slots-claimed': _("You can not create a slot because this activity already has the maximum number of allowed slots"),
+        'user-slot-creation-denied': _("You can not create slots on this activity. I honestly don't know why I'm even showing you this option.")
+    }
+
+    def get_form_kwargs(self):
+        kwargs = {
+            'activity': self.activity,
+            'recurrence_id': self.recurrence_id,
+            'user': self.request.user,
+        }
+        kwargs.update(super(ActivityMomentWithSlotsView, self).get_form_kwargs())
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+
+        # Determine if the mode allows slot creation. If mode is None, staff users should be able to create slots
+        # Note that actual validation always takes place in the form itself, this is merely whether, without any actual
+        # input on the complex database status (i.e. relations), this button needs to be shown.
+        if self.activity.slot_creation == "CREATION_USER":
+            new_slot_form = RegisterNewSlotForm(
+                initial={
+                    'sign_up': True,
+                },
+                activity=self.activity,
+                user=self.request.user,
+                recurrence_id=self.recurrence_id,
+            )
+        elif self.activity.slot_creation == "CREATION_NONE" and self.request.user.is_staff:
```

Comment:
Earlier on there was a superuser-check rather than staff check.
Moreover, staff members are not necessarily people who should also be able to interact with the activities. For instance, at the moment there is the 'Achievement Manager' role, which gives someone access to the achievements and claimants in the admin panel (and as such makes them a staff member).

(The same "TODO Permission system" comment as earlier also applies here)